test/test_datapipe.py

```diff
@@ -776,6 +782,13 @@ def test_demux_datapipe(self):
         with self.assertRaises(ValueError):
             next(it1)
 
+        # Test Case: __len__ not implemented
+        dp1, dp2 = input_dp.demux(num_instances=2, classifier_fn=lambda x: x % 2)
+        with self.assertRaises(TypeError):
```

Comment:
IIRC, this is going to be `NotImplementedError`

Reply:
I think Python throws a `TypeError` when `len()` is not implemented.

```
test/test_datapipe.py:792 (TestFunctionalIterDataPipe.test_error_some)
Traceback (most recent call last):
  File "/Users/ktse/pytorch/test/test_datapipe.py", line 802, in test_error_some
    len(dp1)
  File "/Users/ktse/pytorch/torch/utils/data/datapipes/iter/combining.py", line 162, in __len__
    return len(self.main_datapipe)
TypeError: object of type '_DemultiplexerIterDataPipe' has no len()

```